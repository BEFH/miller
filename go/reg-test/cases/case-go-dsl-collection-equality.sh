run_mlr -n put 'end { @eq = [1,2,3]       == [1,2,3]       ; print @eq}'
run_mlr -n put 'end { @eq = [1,2,3]       == [1,2,3,4]     ; print @eq}'
run_mlr -n put 'end { @eq = [1,2,3]       == [1,3,3]       ; print @eq}'
run_mlr -n put 'end { @eq = ["a",2,3]     == [1,2,3]       ; print @eq}'
run_mlr -n put 'end { @eq = []            == {}            ; print @eq}'
run_mlr -n put 'end { @eq = {}            == {}            ; print @eq}'
run_mlr -n put 'end { @eq = {"a":1}       == {"a":1}       ; print @eq}'
run_mlr -n put 'end { @eq = {"a":1}       == {"a":2}       ; print @eq}'
run_mlr -n put 'end { @eq = {"a":1}       == {"b":1}       ; print @eq}'
run_mlr -n put 'end { @eq = {"a":1,"b":2} == {"b":2}       ; print @eq}'
run_mlr -n put 'end { @eq = {"a":1,"b":2} == {"a":1,"b":2} ; print @eq}'
run_mlr -n put 'end { @eq = {"b":2,"a":1} == {"a":1,"b":2} ; print @eq}'
