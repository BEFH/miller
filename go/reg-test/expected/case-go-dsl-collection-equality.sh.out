
mlr -n put end { @eq = [1,2,3]       == [1,2,3]       ; print @eq}
true

mlr -n put end { @eq = [1,2,3]       == [1,2,3,4]     ; print @eq}
false

mlr -n put end { @eq = [1,2,3]       == [1,3,3]       ; print @eq}
false

mlr -n put end { @eq = ["a",2,3]     == [1,2,3]       ; print @eq}
false

mlr -n put end { @eq = []            == {}            ; print @eq}
false

mlr -n put end { @eq = {}            == {}            ; print @eq}
true

mlr -n put end { @eq = {"a":1}       == {"a":1}       ; print @eq}
true

mlr -n put end { @eq = {"a":1}       == {"a":2}       ; print @eq}
false

mlr -n put end { @eq = {"a":1}       == {"b":1}       ; print @eq}
false

mlr -n put end { @eq = {"a":1,"b":2} == {"b":2}       ; print @eq}
false

mlr -n put end { @eq = {"a":1,"b":2} == {"a":1,"b":2} ; print @eq}
true

mlr -n put end { @eq = {"b":2,"a":1} == {"a":1,"b":2} ; print @eq}
true

